### ISSUE TITLE
Improved handling of fixme

### ISSUE THREAD

**Issue by avylove** [pylint-dev/pylint#8734]
### Current problem

A common policy is that TODO comments are not allowed in code unless they reference an existing issue. Now to fully validate this is out of scope, but it should be _relatively_ easy to implement a check using a regex. However, the implementation today is not obvious and kind of a hack.

For example, if format a valid TODO is
`# TODO (Issue 40): This bird is no more`
And an invalid TODO is
 `# TODO: This bird is no more`

Today you would implement this similarly to

```yaml
[tool.pylint.miscellaneous]
# List of note tags to take in consideration, separated by a comma
notes=["SomethingThatWillNotBeFound"]

# Regular expression of note tags to take in consideration.
notes-rgx="TODO(?! \\(Issue \\d+\\))"
```

`notes` has to have something that won't be found because, if it is left empty, it won't find anything and if left as the default, it will flag all the TODO lines.

### Desired solution

I'm not sure what the ideal solution is. I can think of 3 possible solutions:

1. When `notes` is empty, don't use it to build the regex
2. When `notes-rgx` is given, ignore `notes`
3. Add an additional regex or setting that can be used to to detect this specific case. Ex: `ignore-notes-with-issue`, `notes-issue-rgx`

Clearly, `1` is the easiest to implement and should probably be done regardless.
`2` is a breaking change, but would probably be more intuitive.
`3` really depends on how common a use case this is. For me, it's been the policy at most places I've worked and we ended up disabling this check and rely on code reviews to catch it. It was only recently I thought about tweaking the check.

**Comment by Pierre-Sassoulas** [comment]
This already exists, it's ``notes-rgx``, see https://pylint.readthedocs.io/en/stable/user_guide/configuration/all-options.html#notes-rgx

**Comment by Pierre-Sassoulas** [comment]
I'm in favor of ``when notes-rgx is given, ignore notes``, it's what make the most sense imo.
